From 1ca6fcf940bc306d59651deddf58f155b9feb983 Mon Sep 17 00:00:00 2001
From: David Mohammed <fossfreedom@users.noreply.github.com>
Date: Sun, 12 Jan 2025 09:39:05 +0000
Subject: [PATCH] Add our wayland compatible screenlocker (#636)

* Add our wayland compatible screenlocker

* Support gtklock as a screen-locker

* Use upower library rather than upower cli for calculating discharging vs on AC power

* Define upower dependency and version at the top-level meson.build

* Add build option to check for runtime dependencies (default)
This is to aid packagers/developers where there are apps
that are needed to successfully run budgie-desktop.
For the final packaging, distros can switch off this check
to speed up build-time i.e. limit the number of additional
dependencies these packages would bring in.

* Add build option for the CI

* Make runtime-dependencies option alphabetical in the options list

* Reformat to use tab indentation; add additional comments for options and switch to using dbus rather than systemctl for suspend
---
 .github/workflows/build.yml                   |   2 +-
 meson.build                                   |  23 +-
 meson_options.txt                             |   2 +-
 src/daemon/manager.vala                       |   6 +
 src/daemon/meson.build                        |   2 +
 src/daemon/screenlock.vala                    | 324 ++++++++++++++++++
 src/dialogs/power/dbus.vala                   |   2 +-
 src/dialogs/power/window.vala                 |  17 +-
 .../user-indicator/UserIndicatorWindow.vala   |  17 +-
 ...ie-desktop-screensaver-gnome.desktop.in.in |  14 -
 .../budgie-desktop-screensaver.desktop.in.in  |  14 -
 src/session/meson.build                       |  24 --
 12 files changed, 351 insertions(+), 96 deletions(-)
 create mode 100644 src/daemon/screenlock.vala
 delete mode 100644 src/session/budgie-desktop-screensaver-gnome.desktop.in.in
 delete mode 100644 src/session/budgie-desktop-screensaver.desktop.in.in

diff --git a/.github/workflows/build.yml b/.github/workflows/build.yml
index f885588bc..b659bc283 100644
--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -51,5 +51,5 @@ jobs:
           submodules: true
       - name: Build Budgie Desktop
         run: |
-          meson build -Dci=true
+          meson build -Dci=true -Dwith-runtime-dependencies=false
           meson compile -C build
diff --git a/meson.build b/meson.build
index 0bb250f11..22c5301e9 100644
--- a/meson.build
+++ b/meson.build
@@ -22,17 +22,6 @@ am_cflags = [
 ]
 add_global_arguments(am_cflags, language: 'c')
 
-budgie_screensaver  = find_program('budgie-screensaver', required: false)
-gnome_screensaver = find_program('gnome-screensaver', required: false)
-with_gnome_screensaver = get_option('with-gnome-screensaver')
-if with_gnome_screensaver == false
-    assert(budgie_screensaver.found(), 'Budgie Screensaver is a required separate component of Budgie Desktop. Please ensure budgie-screensaver is available at configure and run time. You can find this at https://github.com/BuddiesOfBudgie/budgie-screensaver')
-    budgie_screensaver_vala_args = []
-else
-    assert(gnome_screensaver.found(), 'Either GNOME Screensaver or Budgie Screensaver is a required separate component of Budgie Desktop. Please ensure one of these is available at configure and run time.')
-    budgie_screensaver_vala_args = ['-D', 'HAVE_GNOME_SCREENSAVER']
-endif
-
 meson.add_install_script('scripts/mesonPostInstall.sh')
 
 fs = import('fs')
@@ -75,9 +64,21 @@ dep_canberra_gtk3 = dependency('libcanberra-gtk3')
 # Needed for window tracking
 dep_xfce4windowing = dependency('libxfce4windowing-0', version: '>= 4.19.7')
 
+# Needed for upower interaction
+dep_upower = dependency('upower-glib', version: '>= 0.99.8')
+
 # Create config.h
 cdata = configuration_data()
 
+if get_option('with-runtime-dependencies')
+    find_program('wlopm', required: true)
+    find_program('swayidle', required: true)
+    found_gtklock = find_program('gtklock', required: false)
+    if found_gtklock.found() == false
+        find_program('swaylock', required: true)
+    endif
+endif
+
 # Inspired by https://github.com/GNOME/recipes/blob/master/meson.build
 package_version = meson.project_version()
 
diff --git a/meson_options.txt b/meson_options.txt
index 223152a51..585e23cdf 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -1,10 +1,10 @@
 option('ci', type: 'boolean', value: false, description: 'Continuous Integration flag (internal use only)')
 option('use-old-zenity', type: 'boolean', value: false, description: 'Use old zenity CLI API for out-of-process dialog handling')
 option('with-bluetooth', type: 'boolean', value: true, description: 'Enable Bluetooth (Vala option)')
-option('with-gnome-screensaver', type: 'boolean', value: false, description: 'Build using gnome-screensaver as a dependency')
 option('with-gtk-doc', type: 'boolean', value: true, description: 'Build gtk-doc documentation')
 option('with-hibernate', type: 'boolean', value: true, description: 'Include support for system hibernation')
 option('with-libuuid-time-safe', type: 'boolean', value: true, description: 'Enable use of LIBUUID.generate_time_safe (Vala option)')
 option('with-polkit', type: 'boolean', value: true, description: 'Enable PolKit support')
+option('with-runtime-dependencies', type: 'boolean', value: true, description: 'Check during build for critical runtime dependencies')
 option('with-stateless', type: 'boolean', value: false, description: 'Enable stateless XDG paths')
 option('xdg-appdir', type: 'string', description: 'XDG autostart path')
diff --git a/src/daemon/manager.vala b/src/daemon/manager.vala
index 9da8c391d..7be675ec2 100644
--- a/src/daemon/manager.vala
+++ b/src/daemon/manager.vala
@@ -28,6 +28,8 @@ namespace Budgie {
 		Budgie.XDGDirTracker? xdg_tracker;
 		Budgie.Background? background;
 
+		/* Screenlock */
+		Budgie.Screenlock? screenlock;
 		/**
 		* Construct a new ServiceManager and initialiase appropriately
 		*/
@@ -58,6 +60,10 @@ namespace Budgie {
 			}
 			xdg_tracker = new Budgie.XDGDirTracker();
 			xdg_tracker.setup_dbus(replace);
+
+
+			screenlock = Screenlock.init();
+			screenlock.setup_dbus();
 		}
 
 		/**
diff --git a/src/daemon/meson.build b/src/daemon/meson.build
index 27e8fd00b..9ebec5cdc 100644
--- a/src/daemon/meson.build
+++ b/src/daemon/meson.build
@@ -35,6 +35,7 @@ daemon_sources = [
     'menus.vala',
     'osd.vala',
     'settings.vala',
+    'screenlock.vala',
     'tabswitcher.vala',
     'xdgdirtracker/dbus.vala',
 ]
@@ -51,6 +52,7 @@ daemon_deps = [
     dep_xfce4windowing,
     dep_gst,
     dep_cairo,
+    dep_upower,
     link_libbudgieprivate,
     libpanelplugin_vapi,
     link_libappsys,
diff --git a/src/daemon/screenlock.vala b/src/daemon/screenlock.vala
new file mode 100644
index 000000000..cd5a9c8d2
--- /dev/null
+++ b/src/daemon/screenlock.vala
@@ -0,0 +1,324 @@
+/*
+ * This file is part of budgie-desktop
+ *
+ * Copyright Budgie Desktop Developers
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ */
+
+namespace Budgie {
+	const string DBUS_SCREENLOCK = "org.buddiesofbudgie.BudgieScreenlock";
+	const string DBUS_SCREENLOCK_PATH = "/org/buddiesofbudgie/Screenlock";
+
+	[DBus (name="org.buddiesofbudgie.BudgieScreenlock")]
+	public class Screenlock {
+		static Screenlock? instance;
+
+		// connections to various schemas used in screenlocking
+		private GLib.Settings power;
+		private GLib.Settings session;
+		private GLib.Settings screensaver;
+		private GLib.Settings lockdown;
+
+		// public functions should check this variable first and exit gracefully where needed
+		bool all_apps = true;
+
+		// remember the locker to be used
+		private string locker = "";
+
+		/** Our upower client */
+		private Up.Client client;
+		private GLib.GenericArray<Up.Device> devices;
+		private bool battery_mode;
+
+		private string calculate_lockcommand() {
+			string output = "";
+
+			if (locker == "swaylock") {
+				// grab the background picture and strip the file prefix
+				string current_wallpaper = screensaver.get_string("picture-uri").replace("file://","");
+
+				// swaylock interprets : as a delimiter for the output name
+				// recommendation by swaylock is to replace : with ::
+				current_wallpaper = current_wallpaper.replace(":", "::");
+
+				File file = File.new_for_path(current_wallpaper);
+				if (current_wallpaper == "" || !file.query_exists()) {
+					// -c Turn the screen into the given color  in this case black
+					output = "-c 000000";
+				}
+				else {
+					// -i display the given image as the current wallpaper
+					output = "-i " + current_wallpaper;
+				}
+
+				/*  -F Show current count of failed attempts
+					-k Show the xkb layout while typing
+					-l Show the Caps Lock state
+					-f daemonize swaylock
+				 */
+				output = "swaylock -Fklf " + output;
+			}
+			else if (locker == "gtklock") {
+				// -d daemonize gtklock
+				output = "gtklock -d";
+
+				/**
+				* Try in order, and load the first one that exists:
+				* - /etc/budgie-desktop/[gtklock.ini | gtklock.css]
+				* - /usr/share/budgie-desktop/[gtklock.ini | gtklock.css]
+				*/
+				string[] system_configs = {
+					@"file://$(Budgie.CONFDIR)/budgie-desktop/gtklock.ini",
+					@"file://$(Budgie.DATADIR)/budgie-desktop/gtklock.ini"
+				};
+
+				foreach (string? filepath in system_configs) {
+					File file = File.new_for_uri(filepath);
+					bool tmp = file.query_exists();
+					if (tmp) {
+						// -c load the config file
+						output += " -c " + file.get_path();
+						break;
+					}
+				}
+
+				string[] style_configs = {
+					@"file://$(Budgie.CONFDIR)/budgie-desktop/gtklock.css",
+					@"file://$(Budgie.DATADIR)/budgie-desktop/gtklock.css"
+				};
+
+				foreach (string? filepath in style_configs) {
+					File file = File.new_for_uri(filepath);
+					bool tmp = file.query_exists();
+					if (tmp) {
+						// -s load the style file
+						output += " -s " + file.get_path();
+						break;
+					}
+				}
+			}
+
+			return output;
+		}
+
+		private string calculate_sleep() {
+			string output = "";
+
+			// number of seconds on ac power that is inactive before action taken; 0 is never
+			int sleep_inactive_ac_timeout = this.power.get_int("sleep-inactive-ac-timeout");
+
+			// BCC sets "nothing" or "suspend" when toggled
+			string sleep_inactive_ac_type = this.power.get_string("sleep-inactive-ac-type");
+
+			// number of seconds on battery that is inactive before action taken; 0 is never
+			int sleep_inactive_battery_timeout = this.power.get_int("sleep-inactive-battery-timeout");
+
+			// BCC sets "nothing" or "suspend" when toggled
+			string sleep_inactive_battery_type = this.power.get_string("sleep-inactive-battery-type");
+
+			string suspend = "dbus-send --system --print-reply --dest=org.freedesktop.login1 /org/freedesktop/login1 org.freedesktop.login1.Manager.Suspend boolean:false";
+			if (sleep_inactive_battery_type == "suspend" && sleep_inactive_battery_timeout != 0) {
+				output = "timeout " + sleep_inactive_battery_timeout.to_string();
+				output += " 'if dbus-send --print-reply=literal --dest=org.buddiesofbudgie.BudgieScreenlock /org/buddiesofbudgie/Screenlock org.buddiesofbudgie.BudgieScreenlock.OnBattery | grep \"boolean true\" > /dev/null; then ";
+				output += suspend + "; fi' ";
+			}
+
+			if (sleep_inactive_ac_type == "suspend" && sleep_inactive_ac_timeout !=0) {
+				output += " timeout " + sleep_inactive_ac_timeout.to_string();
+				output += " 'if dbus-send --print-reply=literal --dest=org.buddiesofbudgie.BudgieScreenlock /org/buddiesofbudgie/Screenlock org.buddiesofbudgie.BudgieScreenlock.OnBattery | grep \"boolean false\" > /dev/null; then ";
+				output += suspend + "; fi' ";
+			}
+
+			return output;
+		}
+
+		private void calculate_idle() {
+			uint idle_delay = 0;
+
+			string new_idle = "";
+
+			bool lock_enabled = false;
+			uint lock_delay = 0;
+
+			bool disable_lock_screen = false;
+
+			idle_delay = this.session.get_uint("idle-delay");
+
+			lock_enabled = this.screensaver.get_boolean("lock-enabled");
+			lock_delay = this.screensaver.get_uint("lock-delay");
+
+			disable_lock_screen = this.lockdown.get_boolean("disable-lock-screen");
+
+			new_idle = "swayidle -w ";
+
+			if (idle_delay !=0 ) {
+				new_idle += "timeout " + idle_delay.to_string() + " 'wlopm --off \\*' resume 'wlopm --on \\*' ";
+			}
+
+			if (lock_enabled && !disable_lock_screen) {
+				new_idle += "timeout " + (idle_delay + lock_delay).to_string() + " '" + calculate_lockcommand() + "' ";
+			}
+
+			if (!disable_lock_screen) {
+				new_idle += "before-sleep '" + calculate_lockcommand() + "'";
+			}
+
+			new_idle += " " + calculate_sleep();
+
+			debug("%s", new_idle);
+			try {
+				string[] spawn_args = {"killall", "swayidle"};
+				string[] spawn_env = Environ.get();
+
+				/* sometimes swayidle exists twice - so we run the kill Process
+				   to make sure before starting swayidle again
+				*/
+				for (var i = 0; i <= 1; i++) {
+					Process.spawn_sync ("/",
+								spawn_args,
+								spawn_env,
+								SpawnFlags.SEARCH_PATH |
+								SpawnFlags.STDERR_TO_DEV_NULL |
+								SpawnFlags.STDOUT_TO_DEV_NULL,
+								null,
+								null,
+								null,
+								null);
+				}
+
+				Process.spawn_command_line_async(new_idle);
+			} catch (SpawnError e) {
+				print("Error: %s\n", e.message);
+			}
+		}
+
+		[DBus (visible = false)]
+		public void setup_dbus() {
+			/* Hook up screenlock dbus */
+			Bus.own_name(BusType.SESSION, DBUS_SCREENLOCK, BusNameOwnerFlags.REPLACE,
+				on_bus_acquired,
+				() => {},
+				() => {} );
+		}
+
+		void on_bus_acquired(DBusConnection conn) {
+			try {
+				conn.register_object(DBUS_SCREENLOCK_PATH, this);
+			} catch (Error e) {
+				message("Unable to register Screenlock: %s", e.message);
+			}
+		}
+
+		public async void lock() throws GLib.DBusError, GLib.IOError {
+			if (!all_apps) {
+				yield;
+				return;
+			}
+
+			try {
+				Process.spawn_command_line_async(calculate_lockcommand());
+			} catch (SpawnError e) {
+				print("Error: %s\n", e.message);
+			}
+
+			yield;
+		}
+
+		[DBus (visible = false)]
+		public static unowned Screenlock init() {
+			if (instance == null)
+				instance = new Screenlock();
+
+			return instance;
+		}
+
+		public bool on_battery() throws GLib.Error {
+			return battery_mode;
+		}
+
+		private void client_daemon(GLib.Object obj, GLib.ParamSpec? sp) {
+			battery_mode = client.get_on_battery();
+		}
+
+		private Screenlock() {
+			string check_apps[] = {"swayidle", "killall", "wlopm", "dbus-send"};
+
+			foreach(unowned var app in check_apps) {
+				if (Environment.find_program_in_path(app) == null) {
+					warning(app + " is not found for screenlocking");
+					all_apps = false;
+				}
+			}
+
+			string supported_lockers[] = {"gtklock", "swaylock"};
+
+
+			foreach(unowned var app in supported_lockers) {
+				if (Environment.find_program_in_path(app) != null) {
+					locker = app;
+					break;
+				}
+			}
+
+			if (locker == "") {
+				warning("No supported screen-locker has been found");
+			}
+
+			if (!all_apps || locker == "") {
+				return;
+			}
+
+			// Connect to upower and get notifications for all batteries and power-supplies to see if on battery mode
+			client = new Up.Client();
+			client.notify.connect(this.client_daemon);
+			devices = client.get_devices2();
+			foreach (var device in devices) {
+				if (device.kind == Up.DeviceKind.LINE_POWER || device.kind == Up.DeviceKind.BATTERY) {
+					device.notify.connect(this.client_daemon);
+				}
+			}
+
+			battery_mode = client.get_on_battery();
+
+			this.power = new Settings("org.gnome.settings-daemon.plugins.power");
+			this.power.changed.connect((key) => {
+				string[] search = { "sleep-inactive-ac-timeout",
+									"sleep-inactive-ac-type",
+									"sleep-inactive-battery-timeout",
+									"sleep-inactive-battery-type"};
+
+				if (key in search) {
+					calculate_idle();
+				}
+			});
+
+			this.session = new Settings("org.gnome.desktop.session");
+			this.session.changed.connect((key) => {
+				if (key == "idle-delay") {
+					calculate_idle();
+				}
+			});
+
+			this.screensaver = new Settings("org.gnome.desktop.screensaver");
+			this.screensaver.changed.connect((key) => {
+				string[] search = {"lock-enabled", "lock-delay", "picture-uri"};
+
+				if (key in search) {
+					calculate_idle();
+				}
+			});
+			this.lockdown = new Settings("org.gnome.desktop.lockdown");
+			this.lockdown.changed.connect((key) => {
+				if (key == "disable-lock-screen") {
+					calculate_idle();
+				}
+			});
+
+			calculate_idle();
+		}
+	}
+}
diff --git a/src/dialogs/power/dbus.vala b/src/dialogs/power/dbus.vala
index d168d7587..2730b946f 100644
--- a/src/dialogs/power/dbus.vala
+++ b/src/dialogs/power/dbus.vala
@@ -40,7 +40,7 @@ namespace Budgie {
 		public abstract async void Shutdown() throws Error;
 	}
 
-	[DBus (name="org.gnome.ScreenSaver")]
+	[DBus (name="org.buddiesofbudgie.BudgieScreenlock")]
 	public interface ScreensaverRemote : GLib.Object {
 		public abstract void lock() throws Error;
 	}
diff --git a/src/dialogs/power/window.vala b/src/dialogs/power/window.vala
index b3f31f271..447cf856a 100644
--- a/src/dialogs/power/window.vala
+++ b/src/dialogs/power/window.vala
@@ -182,14 +182,9 @@ namespace Budgie {
 			}
 
 			try {
-				screensaver = yield Bus.get_proxy(BusType.SESSION, "org.gnome.ScreenSaver", "/org/gnome/ScreenSaver");
+				screensaver = yield Bus.get_proxy(BusType.SESSION, "org.buddiesofbudgie.BudgieScreenlock", "/org/buddiesofbudgie/Screenlock");
 			} catch (Error e) {
-#if HAVE_GNOME_SCREENSAVER
-				warning("Unable to connect to gnome-screensaver: %s", e.message);
-#else
-				warning("Unable to connect to budgie-screensaver: %s", e.message);
-#endif
-				return;
+				warning("Unable to connect to budgie-screenlock: %s", e.message);
 			}
 
 			try {
@@ -296,17 +291,9 @@ namespace Budgie {
 			hide();
 			Idle.add(() => {
 				try {
-#if HAVE_GNOME_SCREENSAVER
-					if (screensaver == null) { // attempt to connect to dbus if not started previously
-						screensaver = Bus.get_proxy_sync(BusType.SESSION, "org.gnome.ScreenSaver", "/org/gnome/ScreenSaver");
-					}
-#endif
 					screensaver.lock();
 				} catch (Error e) {
 					warning("Cannot lock screen: %s", e.message);
-#if HAVE_GNOME_SCREENSAVER
-					screensaver = null; // allow another retry to lock the screen on a failure
-#endif
 				}
 				return false;
 			});
diff --git a/src/panel/applets/user-indicator/UserIndicatorWindow.vala b/src/panel/applets/user-indicator/UserIndicatorWindow.vala
index 6b9e5d744..eb230277e 100644
--- a/src/panel/applets/user-indicator/UserIndicatorWindow.vala
+++ b/src/panel/applets/user-indicator/UserIndicatorWindow.vala
@@ -67,14 +67,9 @@ public class UserIndicatorWindow : Budgie.Popover {
 		}
 
 		try {
-			saver = yield Bus.get_proxy(BusType.SESSION, "org.gnome.ScreenSaver", "/org/gnome/ScreenSaver");
+			saver = yield Bus.get_proxy(BusType.SESSION, "org.buddiesofbudgie.BudgieScreenlock", "/org/buddiesofbudgie/Screenlock");
 		} catch (Error e) {
-#if HAVE_GNOME_SCREENSAVER
-			warning(UNABLE_CONTACT + "gnome-screensaver: %s", e.message);
-#else
-			warning(UNABLE_CONTACT + "budgie-screensaver: %s", e.message);
-#endif
-			return;
+			warning(UNABLE_CONTACT + "budgie-screenlock: %s", e.message);
 		}
 
 		try {
@@ -369,17 +364,9 @@ public class UserIndicatorWindow : Budgie.Popover {
 		hide();
 		Idle.add(() => {
 			try {
-#if HAVE_GNOME_SCREENSAVER
-				if (saver == null) { // attempt to connect to dbus if not started previously
-					saver = Bus.get_proxy_sync(BusType.SESSION, "org.gnome.ScreenSaver", "/org/gnome/ScreenSaver");
-				}
-#endif
 				saver.lock();
 			} catch (Error e) {
 				warning("Cannot lock screen: %s", e.message);
-#if HAVE_GNOME_SCREENSAVER
-				saver = null; // allow another retry to lock the screen on a failure
-#endif
 			}
 			return false;
 		});
diff --git a/src/session/budgie-desktop-screensaver-gnome.desktop.in.in b/src/session/budgie-desktop-screensaver-gnome.desktop.in.in
deleted file mode 100644
index 490f97d86..000000000
--- a/src/session/budgie-desktop-screensaver-gnome.desktop.in.in
+++ /dev/null
@@ -1,14 +0,0 @@
-[Desktop Entry]
-Type=Application
-Name=Screensaver (Budgie Desktop)
-Comment=Launch screensaver and locker program
-Icon=preferences-desktop-screensaver
-Exec=gnome-screensaver
-TryExec=gnome-screensaver
-OnlyShowIn=Budgie;
-NoDisplay=true
-X-GNOME-Autostart-Phase=Application
-X-GNOME-Autostart-Notify=true
-X-GNOME-Bugzilla-Bugzilla=GNOME
-X-GNOME-Bugzilla-Product=gnome-screensaver
-X-GNOME-Bugzilla-Component=general
diff --git a/src/session/budgie-desktop-screensaver.desktop.in.in b/src/session/budgie-desktop-screensaver.desktop.in.in
deleted file mode 100644
index fd9491f53..000000000
--- a/src/session/budgie-desktop-screensaver.desktop.in.in
+++ /dev/null
@@ -1,14 +0,0 @@
-[Desktop Entry]
-Type=Application
-Name=Screensaver (Budgie Desktop)
-Comment=Launch screensaver and locker program
-Icon=preferences-desktop-screensaver
-Exec=budgie-screensaver
-TryExec=budgie-screensaver
-OnlyShowIn=Budgie;
-NoDisplay=true
-X-GNOME-Autostart-Phase=Application
-X-GNOME-Autostart-Notify=true
-X-GNOME-Bugzilla-Bugzilla=GNOME
-X-GNOME-Bugzilla-Product=budgie-screensaver
-X-GNOME-Bugzilla-Component=general
diff --git a/src/session/meson.build b/src/session/meson.build
index b758a72fa..31dd1a465 100644
--- a/src/session/meson.build
+++ b/src/session/meson.build
@@ -103,30 +103,6 @@ custom_target('desktop-file-nm-applet',
     install_dir : xdg_appdir,
 )
 
-# Write the budgie-desktop-screensaver.in file from the appropriate .in.in
-if with_gnome_screensaver == true
-    session_screensaver = configure_file(
-        input: 'budgie-desktop-screensaver-gnome.desktop.in.in',
-        output: 'budgie-desktop-screensaver.desktop.in',
-        configuration: session_data,
-    )
-else
-    session_screensaver = configure_file(
-        input: 'budgie-desktop-screensaver.desktop.in.in',
-        output: 'budgie-desktop-screensaver.desktop.in',
-        configuration: session_data,
-    )
-endif
-
-# Merge + install screensaver
-custom_target('desktop-file-screensaver',
-    input : session_screensaver,
-    output : 'org.buddiesofbudgie.BudgieDesktopScreensaver.desktop',
-    command : [intltool, '--desktop-style', podir, '@INPUT@', '@OUTPUT@'],
-    install : true,
-    install_dir : xdg_appdir,
-)
-
 # Merge + install power dialog
 session_power_dialog_desktop = configure_file(
     input: 'budgie-desktop-powerdialog.desktop.in.in',
